docs/en/sql-reference/functions/array-functions.md

```diff
@@ -2118,6 +2118,80 @@ Result:
 └─────────────────────┘
 ```
 
+
+---
+
+## arrayRandomSample
+
+The `arrayRandomSample` function randomly selects elements from the input array, ensuring that the specified number of elements is included in the sample. If the `samples` exceeds the size of the input array, the sample size is automatically limited to the size of the array. In such cases, all elements of the array may be returned, but the order is not guaranteed. The function can handle both flat arrays and nested arrays.
+
+**Syntax**
+
+```sql
+arrayRandomSample(arr, samples)
+```
+
+**Arguments**
+
+- `arr` — The input array from which to sample elements. This can include both flat arrays (e.g., integers, strings) and nested arrays.
+- `samples` — An unsigned integer specifying the number of elements to include in the random sample.
+
+**Returned Value**
+
+- An array containing a random sample of elements from the input array.
+
+**Examples**
+
+Query:
+
+```sql
+SELECT arrayRandomSample(['apple', 'banana', 'cherry', 'date'], 2) as res;
+```
+
+Result:
+```
+┌─res────────────────┐
+│ ['banana','apple'] │
+└────────────────────┘
+```
+
+Query:
+
+```sql
+SELECT arrayRandomSample([[1, 2], [3, 4], [5, 6]], 2) as res;
+```
+
+Result:
+```
+┌─res───────────┐
+│ [[3,4],[5,6]] │
+└───────────────┘
+```
+
+Query:
+
+```sql
+SELECT arrayRandomSample([1, 2, 3, 4, 5], 0) as res;
+```
+
+Result:
+```
+TODO: FIX ME
```

Comment:
(still says FIX ME)

Reply:
Yep, please give me a couple of minutes. I will fix everything.